src/Data/Replicator/Linear/Internal.hs

```diff
@@ -133,13 +133,26 @@ extend f = map f . duplicate
 --
 -- > elim (,) :: Replicator a %1 -> (a, a)
 -- > elim (,,) :: Replicator a %1 -> (a, a, a)
-elim :: forall (n :: Nat) a b f. (Elim (NatToPeano n) a b, IsFunN a b f, f ~ FunN (NatToPeano n) a b, n ~ Arity b f) => f %1 -> Replicator a %1 -> b
+elim ::
+  forall (n :: Nat) a b f.
+  ( -- Always fulfilled. Provides the actual implementation of 'elim' in
+    -- terms of @elim\'@
+    Elim (NatToPeano n) a b,
+    -- Asserts that @f@ is a function taking @n@ linear arguments of type
+    -- @a@ and returning a value of type @b@
+    IsFunN a b f,
```

Comment:
I'm not sure it's worth repeating the documentation of `IsFunN` here.

Reply:
The comment was targeting the three last constraints `IsFunN a b f, f ~ FunN (NatToPeano n) a b, n ~ Arity b f`, but Ormolu made them multiline :(

At the beginning, I added these comments for the end-user, but because haddock can't accept comments on constraints, maybe it's worth to remove them all.

Reply:
You can always write text in the docstring. Not everything needs to be structured :slightly_smiling_face: .